### DISCUSSION TITLE
How to find images that is not referenced in any note?

### DISCUSSION THREAD

**Discussion by Nriver** [TriliumNext/Trilium#3135] [Q&A]
When delete a image in the note with `Backspace`, the original image file still exists as a sub note. Or sometimes replacing an image with a better quality one will result in this.

I seldom delete the image note manually, and I suspect there are a lot of such unused/isolated/unreferenced images in my trilium database. How can I find all the images that is not used and delete them? I have thousands of notes in my database, so manual check is not quite possible.

Any suggestion is welcomed.

**Comment by XXXJetfireXXX** [comment]
You can try searching with `note.type = image` as the parameter. Like so:
![image](https://user-images.githubusercontent.com/43646178/190614258-b6c68007-c86f-430d-bde7-9a250a959ccd.png)
Obviously it's gonna show you *all* the images inside your database, but I'm not sure yet how do you filter them by the amount of references.

**Reply by zadam** [reply]
Good start, complete query would be `note.type = image AND note.targetRelationCountIncludingLinks = 0`

**Reply by XXXJetfireXXX** [reply]
Yep, this one does the thing. Thanks zadam!

**Reply by XXXJetfireXXX** [reply]
Also @Nriver, there are more filters that you can take a look at: https://github.com/zadam/trilium/wiki/Search
These might greatly help you with searching stuff!

**Reply by Nriver** [reply]
Problem solved. Thank you all!
